postmaster/static/js/admins.js

```diff
@@ -190,6 +222,39 @@ function adminEventListeners () {
 
     $('#filterRow input').typeWatch(typeWatchOptions);
 
+    twoFAModal.on('show.bs.modal', function (e) {
+      var id = $(e.relatedTarget).data('pk');
+      $.ajax({
+        url: '/api/v1/admins/' + id + '/2factor/qrcode',
+        success: function(data){
+           $("#qrcode").html(new XMLSerializer().serializeToString(data.documentElement));
+           verify2FABtn.attr('data-pk', id);
+        },
+        error: function (response) {
+          addStatusMessage('error', filterText(jQuery.parseJSON(response.responseText).message));
+          twoFAModal.modal('hide');
+        }
+      });
+    });
+
+    verify2FABtn.on('click', function (e) {
+      var id = $(this).attr('data-pk');
+      $.ajax({
+        url: '/api/v1/admins/' + id + '/2factor/verify',
+        method: 'POST',
+        contentType: 'application/json',
+        dataType: 'json',
+        data: JSON.stringify ({code: $("#verify2FACode").val()}),
+        success: function(data){
+          twoFAModal.modal('hide');
+          fillInTable();
+        },
+        error: function (response) {
+          addStatusMessage('error', filterText(jQuery.parseJSON(response.responseText).message));
```

Comment:
Either the modal should be closed on error, or the status message should be in the foreground (the messages are currently behind the modal)